### PR TITLE
Manifest

### PR DESCRIPTION
**Pull request by Talina06**

### CODE REVIEW

**Comment by Talina06** [comment]
@mohandutt134 Please review this PR.

File: commands/cli/layout.go [hunk 1]
```diff
@@ -32,31 +34,24 @@ func (cm *layout) layoutCreate(c *kingpin.ParseContext) error {
...
-			log.Printf("skipping %s", path)
-			return nil
-		}
-
-		files = append(files, path)
-		return nil
+	manifest, err := commons.ReadFileLines(path.Join(cm.dirName, ".tsl8"))
+	if err != nil {
+		log.Println("No manifest file found, moving ahead...")
+	}
```
Comment (jdk2588) [inline commands/cli/layout.go]:
```
if manifest, err := commons.ReadFileLines(path.Join(cm.dirName, ".tsl8")); err != nil {
       log.Println("No manifest file found, moving ahead...")
}
```
Use the error at blocks, we had a discussion around this

File: commands/cli/layout.go [hunk 2]
```diff
@@ -32,31 +34,24 @@ func (cm *layout) layoutCreate(c *kingpin.ParseContext) error {
...
 
-	}); err != nil {
-		return err
+	files, err := commons.CandidateFiles(cm.dirName, manifest)
+	if err != nil {
+		return errors.Wrap(err, "Cannot get files")
```
Comment (jdk2588) [inline commands/cli/layout.go]: Same as above

File: commands/cli/layout.go [hunk 3]
```diff
@@ -32,31 +34,24 @@ func (cm *layout) layoutCreate(c *kingpin.ParseContext) error {
...
 	}
 
+	fmt.Println(files)
```
Comment (jdk2588) [inline commands/cli/layout.go]: Remove println

File: commands/commons/commons.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	file.Read(head)
+
+	fileType, err := filetype.Get(head)
+	return fileType.Extension, nil
+}
+
+func ReadFileLines(file string) ([]string, error) {
+	lines := []string{}
+
+	f, err := os.OpenFile(file, os.O_RDONLY, os.ModePerm)
```
Comment (jdk2588) [inline commands/commons/commons.go]: Check error in single scope

File: commands/cli/layout.go (hunk 1 above)
Comment (jdk2588) [reply commands/cli/layout.go]: Try if can be done this way

Comment (Talina06) [reply commands/cli/layout.go]: This can't be done, as we are returning 2 values neither of which can be ignored.

**Pull request merged**